### PR TITLE
Integrate file_encrypt into file fields.

### PR DESCRIPTION
**Pull request by dawehner**
This PR is build upon #3

This PR allows you to
- Select "encrypt" on file field field storage settings.
- Select an encryption profile on the same page
- Automatically configure the right upload location for those files
- Actually upload the file
- See the actual file on the node itself.

This PR also adds test coverage for
- Creating and configuring fields
- Encrypting file fields

TODO:
- Preview doesn't show the actual image yet [I think this should tackled in a follow up]

### CODE REVIEW

File: file_encrypt.module [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<?php
+use Drupal\Core\Entity\EntityInterface;
```
Comment (nerdstein) [inline file_encrypt.module]: I believe this file should start with some comments

File: file_encrypt.module [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+use Drupal\field\FieldConfigInterface;
+use Drupal\field\FieldStorageConfigInterface;
+use Drupal\file_encrypt\EncryptStreamWrapper;
+use Drupal\file_encrypt\Hooks\FieldStorageConfigEditFormAlter;
+
+/**
+ * Implements hook_form_field_storage_config_edit_form_alter().
+ */
+function file_encrypt_form_field_storage_config_edit_form_alter(&$form, \Drupal\Core\Form\FormStateInterface $form_state) {
+  (new FieldStorageConfigEditFormAlter(\Drupal::service('plugin.manager.field.field_type'), \Drupal::service('encrypt.encryption_profile.manager')))
```
Comment (nerdstein) [inline file_encrypt.module:17]: Wow! That's some shorthand!

File: src/Hooks/FieldStorageConfigEditFormAlter.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
+<?php
+
```
Comment (nerdstein) [inline src/Hooks/FieldStorageConfigEditFormAlter.php:2]: Class comments please

File: src/Hooks/FieldStorageConfigEditFormAlter.php [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+use Drupal\Core\Field\FieldTypePluginManagerInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\encrypt\EncryptionProfileManagerInterface;
+use Drupal\field\FieldStorageConfigInterface;
+use Drupal\file\Plugin\Field\FieldType\FileItem;
+
+class FieldStorageConfigEditFormAlter {
+
+  use StringTranslationTrait;
```
Comment (nerdstein) [inline src/Hooks/FieldStorageConfigEditFormAlter.php:20]: Does this go here or outside of the class? I rarely see use statements inside of classes.

File: src/Hooks/FieldStorageConfigEditFormAlter.php [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            'input[name="settings[uri_scheme]"]' => ['value' => 'encrypt'],
+          ],
+        ],
+      ];
+    }
+
+    $form['#entity_builders'][] = static::class . '::buildEntity';
+  }
+
+  public static function buildEntity($entity_type_id, FieldStorageConfigInterface $field_storage_config, $form, FormStateInterface $form_state) {
```
Comment (nerdstein) [inline src/Hooks/FieldStorageConfigEditFormAlter.php:97]: Comments please

File: src/Hooks/FieldStorageConfigEditFormAlter.php [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    if (is_subclass_of($field_type_definition['class'], FileItem::class)) {
+      $this->doFormAlter($field_storage_config, $form, $form_state);
+    }
+  }
+
+  /**
+   * @param \Drupal\field\FieldStorageConfigInterface $field_storage_config
+   * @param array $form
+   */
+  protected function doFormAlter(FieldStorageConfigInterface $field_storage_config, array &$form) {
```
Comment (nerdstein) [inline src/Hooks/FieldStorageConfigEditFormAlter.php:61]: Should we pass the form state? I generally do that even if I don't immediately need it

File: src/Hooks/FieldStorageConfigEditFormAlter.php [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+   */
+  public function __construct(FieldTypePluginManagerInterface $field_type_manager, EncryptionProfileManagerInterface $encryption_profile_manager) {
+    $this->fieldTypeManager = $field_type_manager;
+    $this->encryptionProfileManager = $encryption_profile_manager;
+  }
+
+  /**
+   * Implements hook_form_field_storage_config_edit_form_alter().
+   */
+  public function alterForm(array &$form, FormStateInterface $form_state) {
```
Comment (nerdstein) [inline src/Hooks/FieldStorageConfigEditFormAlter.php:44]: Should we pass the form state?

File: file_encrypt.module (hunk 1 above)
Comment (dawehner) [inline file_encrypt.module]: File comments are certainly one of the huge problems in tech :)

File: file_encrypt.module (hunk 2 above)
Comment (dawehner) [inline file_encrypt.module:17]: I'm really not sure about this pattern, but I'm using it more and more.

File: src/Hooks/FieldStorageConfigEditFormAlter.php (hunk 3 above)
Comment (dawehner) [inline src/Hooks/FieldStorageConfigEditFormAlter.php:2]: Actually we got rid of them in https://www.drupal.org/node/2304909

File: src/Hooks/FieldStorageConfigEditFormAlter.php (hunk 4 above)
Comment (dawehner) [inline src/Hooks/FieldStorageConfigEditFormAlter.php:20]: Oh, this is a [trait](https://secure.php.net/manual/de/language.oop5.traits.php) , its just how it is

File: src/Hooks/FieldStorageConfigEditFormAlter.php (hunk 7 above)
Comment (dawehner) [inline src/Hooks/FieldStorageConfigEditFormAlter.php:44]: Well, we use it 2 lines below, I doubt we can skip it :)

File: src/Hooks/FieldStorageConfigEditFormAlter.php (hunk 5 above)
Comment (dawehner) [inline src/Hooks/FieldStorageConfigEditFormAlter.php:97]: fair

**Pull request merged**